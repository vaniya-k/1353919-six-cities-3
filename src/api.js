import axios from "axios";

const Error = {
  UNAUTHORIZED: 401
};

export const createAPI = (onUnauthorized) => {
  const api = axios.create({
    baseURL: `https://htmlacademy-react-3.appspot.com/six-cities`,
    timeout: 1000 * 5,
    withCredentials: true,
  });

  const onSuccess = (response) => {
    return response;
  };

  const onFail = (err) => {
    const {response} = err;

    if (response.status === Error.UNAUTHORIZED) {
      onUnauthorized();

      throw err;
    }

    throw err;
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
