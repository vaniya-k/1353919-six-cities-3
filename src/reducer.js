import offers from '../src/mocks/cities-with-places.js';

const firstActiveCity = 0;

const initialState = {
  activeCityId: firstActiveCity,
  activeCityName: offers[firstActiveCity].city,
  places: offers[firstActiveCity].places,
  activeCardLatLon: {lat: null, lon: null},
  activeSortType: 0
};

const ActionType = {
  CHANGE_CITY: `CHANGE_CITY`,
  GET_PLACES: `GET_PLACES`,
  SET_ACTIVE_CARD_LAT_LON: `SET_ACTIVE_CARD_LAT_LON`,
  CHANGE_SORTING: `CHANGE_SORTING`
};

const ActionCreator = {
  changeCity: (cityId) => ({
    type: ActionType.CHANGE_CITY,
    payload: cityId
  }),
  setActiveCardLatLon: (activeCardLatLon) => ({
    type:  ActionType.SET_ACTIVE_CARD_LAT_LON,
    payload: activeCardLatLon
  }),
  changeSorting: (selectedSortType) => ({
    type:  ActionType.CHANGE_SORTING,
    payload: selectedSortType
  })
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return Object.assign({}, state, {
        activeCityId: action.payload,
        activeCityName: offers[action.payload].city,
        places: offers[action.payload].places
      });

    case ActionType.GET_PLACES:
      return state.places;

    case ActionType.SET_ACTIVE_CARD_LAT_LON:
      return Object.assign({}, state, {
        activeCardLatLon: {lat: action.payload.lat, lon: action.payload.lon}
      });
    
    case ActionType.CHANGE_SORTING:
      return Object.assign({}, state, {
        activeSortType: action.payload
      });
  }

  return state;
};

export {reducer, ActionType, ActionCreator};
