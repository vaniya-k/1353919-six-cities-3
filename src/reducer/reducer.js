import {combineReducers} from "redux";
import {reducer as offers} from './offers/offers.js';
import {reducer as user} from './user/user.js';

export default combineReducers({
  offers,
  user,
});
