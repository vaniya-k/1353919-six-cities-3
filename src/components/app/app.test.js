//фейковый пулреквест -- хочу прочитать задание к module8-task2

import React from 'react';
import renderer from 'react-test-renderer';
import App from './app.jsx';
import placesListing from '../../mocks/places-listing-test.js';
import placesFullData from '../../mocks/places-full-data-test.js';
import {createStore, applyMiddleware, compose} from "redux";
import {Provider} from "react-redux";
import thunk from "redux-thunk";
import reducer from "../../reducer/reducer.js";
import {createAPI} from "../../api.js";

it(`<App/> should return the whole page with 54 matches and 3 places displayed`, () => {
  const api = createAPI(() => {});

  const store = createStore(
      reducer,
      compose(
          applyMiddleware(thunk.withExtraArgument(api)),
          window.__REDUX_DEVTOOLS_EXTENSION__ ? window.__REDUX_DEVTOOLS_EXTENSION__() : (f) => f
      )
  );

  const tree = renderer
    .create(<Provider store={store}><App placesListing={placesListing} placePageData={placesFullData[0]}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
