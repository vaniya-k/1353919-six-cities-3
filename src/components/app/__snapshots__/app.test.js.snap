// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should return the whole page with 54 matches and 3 places displayed 1`] = `null`;
